scm-webapp/src/main/java/sonia/scm/plugin/DefaultPluginManager.java

```diff
@@ -143,13 +147,49 @@ public List<InstalledPlugin> getInstalled() {
   @Override
   public List<AvailablePlugin> getAvailable() {
     PluginPermissions.read().check();
-    return center.getAvailable()
+    return center.getAvailablePlugins()
       .stream()
       .filter(this::isNotInstalledOrMoreUpToDate)
       .map(p -> getPending(p.getDescriptor().getInformation().getName()).orElse(p))
       .collect(Collectors.toList());
   }
 
+  @Override
+  public Set<PluginSet> getPluginSets() {
+    PluginPermissions.read().check();
+    return center.getAvailablePluginSets();
+  }
+
+  @Override
+  public void installPluginSets(Set<String> pluginSetIds, boolean restartAfterInstallation) {
+    PluginPermissions.write().check();
+
+    Set<PluginSet> pluginSets = getPluginSets();
+    Set<PluginSet> pluginSetsToInstall = pluginSetIds.stream()
+      .map(id -> pluginSets.stream().filter(pluginSet -> pluginSet.getId().equals(id)).findFirst())
+      .filter(Optional::isPresent)
+      .map(Optional::get)
+      .collect(Collectors.toSet());
+
+    Set<AvailablePlugin> pluginsToInstall = pluginSetsToInstall
+      .stream()
+      .map(pluginSet -> pluginSet
+        .getPlugins()
+        .stream()
+        .map(this::collectPluginsToInstall).flatMap(Collection::stream).collect(Collectors.toSet())
+      )
+      .flatMap(Collection::stream)
+      .collect(Collectors.toSet());
```

Comment:
Why so complicated, what am I missing?
```suggestion
    Set<AvailablePlugin> pluginsToInstall = pluginSetsToInstall
      .stream()
      .flatMap(pluginSet -> pluginSet
        .getPlugins()
        .stream()
        .map(this::collectPluginsToInstall)
        .flatMap(Collection::stream)
      )
      .collect(Collectors.toSet());
```